drivers/led/gpio.c

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <kernel.h>
+#include <led.h>
+#include <gpio.h>
+#include <device.h>
+#include <sys/util.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(led_gpio, CONFIG_LEDS_GPIO_LOG_LEVEL);
+
+#define LED_SW_BLINK BIT(0)
+#define LED_BLINK_DISABLE BIT(1)
+
+struct gpio_cfg {
+	const char *port;
+	u8_t pin;
+	u8_t polarity;
+};
+
+struct led_gpio_data {
+	struct gpio_cfg gpio;
+	const char *label;
+	u32_t id;
+	struct device *gpio_dev;
+#if defined(CONFIG_LED_SOFTWARE_BLINK)
```

Comment:
Please split the software blinking functionality into a separate commit.

Reply:
Ok. I splitted it but why You want me to do it ?

Reply:
Just to make it easy to bisect in future and moreover these 2 are different functionalities.